radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/tests/consensus_ledger_sync/FallBehindMultipleEpochsLedgerSyncTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.integration.distributed.simulation.tests.consensus_ledger_sync;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.multibindings.ProvidesIntoMap;
+import com.google.inject.multibindings.StringMapKey;
+import com.radixdlt.ModuleRunner;
+import com.radixdlt.consensus.bft.BFTValidatorSet;
+import com.radixdlt.consensus.bft.View;
+import com.radixdlt.counters.SystemCounters.CounterType;
+import com.radixdlt.integration.distributed.simulation.NetworkLatencies;
+import com.radixdlt.integration.distributed.simulation.NetworkOrdering;
+import com.radixdlt.integration.distributed.simulation.SimulationTest;
+import com.radixdlt.integration.distributed.simulation.DelayedModuleRunner;
+import com.radixdlt.integration.distributed.simulation.ConsensusMonitors;
+import com.radixdlt.integration.distributed.simulation.LedgerMonitors;
+import com.radixdlt.integration.distributed.simulation.SimulationTest.Builder;
+import com.radixdlt.sync.SyncConfig;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * A node that falls behind multiple epochs should be able to successfully sync its ledger
+ * and keep up once synced.
+ */
+public class FallBehindMultipleEpochsLedgerSyncTest {
+	private static final int NODE_UNDER_TEST_INDEX = 2;
+	private static final long SYNC_DELAY = 5000L;
+
+	private final Builder testBuilder;
+
+	public FallBehindMultipleEpochsLedgerSyncTest() {
+		this.testBuilder = SimulationTest.builder()
+			.numNodes(3)
+			.networkModules(
+				NetworkOrdering.inOrder(),
+				NetworkLatencies.fixed(10)
+			)
+			.addSingleByzantineModule(NODE_UNDER_TEST_INDEX, new AbstractModule() {
+				@ProvidesIntoMap
+				@StringMapKey("sync-delayed")
+				private ModuleRunner delayedSyncRunner(Map<String, ModuleRunner> runners) {
+					return new DelayedModuleRunner(runners.get("sync"), SYNC_DELAY);
```

Comment:
Rather than use `DelayedModuleRunner`, better to call `.start()` on "sync" module after `SYNC_DELAY`